omod/src/main/java/org/openmrs/module/reporting/web/taglib/FormatTag.java

```diff
@@ -323,7 +325,17 @@ private void printDataSet(StringBuilder sb, String title, DataSet dataSet) {
 	    			.append(formatHelper(row.getColumnValue(col)))
 	    			.append("</td></tr>");
 	    	}
-	    } else {
+		} else if (dataSet instanceof IterableSqlDataSet) {
+			DataSetRow row = ((IterableSqlDataSet) dataSet).iterator().next();
+			while (row != null) {
+				sb.append("<tr>");
+				for (DataSetColumn col : cols) {
+					sb.append("<td>").append(formatHelper(row.getColumnValue(col))).append("</td>");
+				}
+				sb.append("</tr>");
+				row = ((IterableSqlDataSet) dataSet).iterator().next();
```

Comment:
Instead of doing the above for each row, isn't it more efficient to do it outside the loop and store the iterator in a variable?

Reply:
done.